### PR TITLE
Add initial test structure.

### PR DESCRIPTION
**Pull request by mpereira**
To run tests: `./tests/run.sh` (or `./run-tests.sh` if you don't have a Go environment set up).

Example run:

``` text
$ ./tests/run.sh
+++ dirname ./tests/run.sh
++ cd ./tests
++ pwd
+ readonly SCRIPT_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/tests
+ SCRIPT_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/tests
++ readlink -f /Users/murilo/mesosphere/kudo-cassandra-operator/tests/..
+ readonly PROJECT_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator
+ PROJECT_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator
+ source /Users/murilo/mesosphere/kudo-cassandra-operator/metadata.sh
++++ dirname /Users/murilo/mesosphere/kudo-cassandra-operator/metadata.sh
+++ cd /Users/murilo/mesosphere/kudo-cassandra-operator
+++ pwd
++ script_directory=/Users/murilo/mesosphere/kudo-cassandra-operator
+++ readlink -f /Users/murilo/mesosphere/kudo-cassandra-operator
++ project_directory=/Users/murilo/mesosphere/kudo-cassandra-operator
++ export PROJECT_NAME=kudo-cassandra-operator
++ PROJECT_NAME=kudo-cassandra-operator
++ export OPERATOR_NAME=cassandra
++ OPERATOR_NAME=cassandra
++ export OPERATOR_VERSION=0.1.0
++ OPERATOR_VERSION=0.1.0
++ export OPERATOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/operator
++ OPERATOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/operator
++ export VENDOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor
++ VENDOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor
++ export CASSANDRA_VERSION=3.11.4
++ CASSANDRA_VERSION=3.11.4
++ export KUDO_VERSION=0.7.4
++ KUDO_VERSION=0.7.4
++ export KUBERNETES_VERSION=1.15.0
++ KUBERNETES_VERSION=1.15.0
++ export CASSANDRA_DOCKER_IMAGE_NAMESPACE=mesosphere
++ CASSANDRA_DOCKER_IMAGE_NAMESPACE=mesosphere
++ export CASSANDRA_DOCKER_IMAGE_NAME=cassandra
++ CASSANDRA_DOCKER_IMAGE_NAME=cassandra
++ export CASSANDRA_DOCKER_IMAGE_TAG=0.1.0-3.11.4
++ CASSANDRA_DOCKER_IMAGE_TAG=0.1.0-3.11.4
++ export CASSANDRA_DOCKER_IMAGE=mesosphere/cassandra:0.1.0-3.11.4
++ CASSANDRA_DOCKER_IMAGE=mesosphere/cassandra:0.1.0-3.11.4
++ export INTEGRATION_TESTS_DOCKER_IMAGE=golang:1.13.1-stretch
++ INTEGRATION_TESTS_DOCKER_IMAGE=golang:1.13.1-stretch
+ export KUBECONFIG=/Users/murilo/.kube/config
+ KUBECONFIG=/Users/murilo/.kube/config
+ export OPERATOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/operator
+ OPERATOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/operator
+ export VENDOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor
+ VENDOR_DIRECTORY=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor
+ export KUBECTL_PATH=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor/kubectl.sh
+ KUBECTL_PATH=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor/kubectl.sh
+ export DS_KUDO_VERSION=v0.7.4
+ DS_KUDO_VERSION=v0.7.4
+ '[' -n '' ']'
+ export GINKGO_PATH=/Users/murilo/go/bin/ginkgo
+ GINKGO_PATH=/Users/murilo/go/bin/ginkgo
+ export PATH=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor:/Users/murilo/.yarn/bin:/Users/murilo/.config/yarn/global/node_modules/.bin:/usr/local/opt/gettext/bin:/usr/local/opt/gnu-getopt/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/opt/X11/bin:/Users/murilo/.yarn/bin:/Users/murilo/.config/yarn/global/node_modules/.bin:/usr/local/opt/gnu-getopt/bin:/usr/local/opt/coreutils/libexec/gnubin:/Applications/Emacs-26.3.app/Contents/MacOS/bin-x86_64-10_14:/Applications/Emacs-26.3.app/Contents/MacOS/libexec-x86_64-10_14:/opt/local/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Applications/CMake.app/Contents/bin:/Users/murilo/Library/Python/3.7/bin:/Library/Frameworks/Python.framework/Versions/3.7/bin:/Users/murilo/.go/bin:/Users/murilo/.cargo/bin:/Applications/MacVim.app/Contents/bin:/Applications/GraalVM-19.0.2.app/Contents/Home/bin:/Users/murilo/mesosphere/bin:/usr/local/bin:/opt/local/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Applications/CMake.app/Contents/bin:/Users/murilo/Library/Python/3.7/bin:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/go/bin:/Users/murilo/.go/bin:/Users/murilo/.cargo/bin:/Applications/MacVim.app/Contents/bin:/Applications/GraalVM-19.0.2.app/Contents/Home/bin:/Users/murilo/mesosphere/bin
+ PATH=/Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor:/Users/murilo/.yarn/bin:/Users/murilo/.config/yarn/global/node_modules/.bin:/usr/local/opt/gettext/bin:/usr/local/opt/gnu-getopt/bin:/usr/local/opt/coreutils/libexec/gnubin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/go/bin:/opt/X11/bin:/Users/murilo/.yarn/bin:/Users/murilo/.config/yarn/global/node_modules/.bin:/usr/local/opt/gnu-getopt/bin:/usr/local/opt/coreutils/libexec/gnubin:/Applications/Emacs-26.3.app/Contents/MacOS/bin-x86_64-10_14:/Applications/Emacs-26.3.app/Contents/MacOS/libexec-x86_64-10_14:/opt/local/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Applications/CMake.app/Contents/bin:/Users/murilo/Library/Python/3.7/bin:/Library/Frameworks/Python.framework/Versions/3.7/bin:/Users/murilo/.go/bin:/Users/murilo/.cargo/bin:/Applications/MacVim.app/Contents/bin:/Applications/GraalVM-19.0.2.app/Contents/Home/bin:/Users/murilo/mesosphere/bin:/usr/local/bin:/opt/local/bin:/Applications/Postgres.app/Contents/Versions/latest/bin:/Applications/CMake.app/Contents/bin:/Users/murilo/Library/Python/3.7/bin:/Library/Frameworks/Python.framework/Versions/3.7/bin:/usr/local/go/bin:/Users/murilo/.go/bin:/Users/murilo/.cargo/bin:/Applications/MacVim.app/Contents/bin:/Applications/GraalVM-19.0.2.app/Contents/Home/bin:/Users/murilo/mesosphere/bin
+ cd /Users/murilo/mesosphere/kudo-cassandra-operator/tests
+ go mod edit -require github.com/kudobuilder/kudo@v0.7.4
+ go install github.com/onsi/ginkgo/ginkgo
+ /Users/murilo/mesosphere/kudo-cassandra-operator/shared/vendor/kubectl.sh kudo version
KUDO Version: version.Info{GitVersion:"0.7.4", GitCommit:"81b14cb3", BuildDate:"2019-10-11T16:44:30Z", GoVersion:"go1.13.1", Compiler:"gc", Platform:"darwin/amd64"}
+ /Users/murilo/go/bin/ginkgo ./suites/...
Running Suite: simple-install-uninstall-test
============================================
Random Seed: 1571923335
Will run 2 of 2 specs

time="2019-10-24T15:22:20+02:00" level=info msg="Configuring kubectl using config file '/Users/murilo/.kube/config' with context ''"
time="2019-10-24T15:22:20+02:00" level=info msg="Using kubeconfig at '/Users/murilo/.kube/config'"
time="2019-10-24T15:22:20+02:00" level=info msg="Uninstalling 'cassandra' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
time="2019-10-24T15:22:25+02:00" level=info msg="exit status: 0\nstdout:\n\nstderr:\nError from server (NotFound): instances.kudo.dev \"cassandra-instance\" not found\nError from server (NotFound): operatorversions.kudo.dev \"cassandra-0.1.0\" not found\nError from server (NotFound): operators.kudo.dev \"cassandra\" not found\n"
time="2019-10-24T15:22:25+02:00" level=info msg="Successfully uninstalled 'cassandra' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
time="2019-10-24T15:22:25+02:00" level=info msg="Creating namespace 'simple-install-uninstall-test-namespace'"
time="2019-10-24T15:22:26+02:00" level=info msg="Created namespace 'simple-install-uninstall-test-namespace'"
time="2019-10-24T15:22:26+02:00" level=info msg="Installing operator from path: '/Users/murilo/mesosphere/kudo-cassandra-operator/operator' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
time="2019-10-24T15:22:30+02:00" level=info msg="exit status: 0\nstdout:\ninstance.kudo.dev/v1alpha1/cassandra-instance created\n\n\nstderr:\n"
time="2019-10-24T15:22:30+02:00" level=info msg="Started operator installation from path: '/Users/murilo/mesosphere/kudo-cassandra-operator/operator' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
time="2019-10-24T15:22:30+02:00" level=info msg="Operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status is 'IN_PROGRESS'"
time="2019-10-24T15:22:31+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:22:41+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:22:51+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:23:01+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:23:11+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:23:21+02:00" level=info msg="Waiting for operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status to be 'COMPLETE', is 'IN_PROGRESS'"
time="2019-10-24T15:23:32+02:00" level=info msg="Operator (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace') deploy status is 'COMPLETE'"
• [SLOW TEST:65.716 seconds]
simple-install-uninstall-test
/Users/murilo/mesosphere/kudo-cassandra-operator/tests/suites/simple_install_uninstall_test.go:35
  Installs the operator from a directory
  /Users/murilo/mesosphere/kudo-cassandra-operator/tests/suites/simple_install_uninstall_test.go:36
------------------------------
time="2019-10-24T15:23:32+02:00" level=info msg="Uninstalling 'cassandra' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
time="2019-10-24T15:23:52+02:00" level=info msg="exit status: 0\nstdout:\ninstance.kudo.dev \"cassandra-instance\" deleted\noperatorversion.kudo.dev \"cassandra-0.1.0\" deleted\noperator.kudo.dev \"cassandra\" deleted\npersistentvolumeclaim \"var-lib-cassandra-cassandra-instance-node-0\" deleted\npersistentvolumeclaim \"var-lib-cassandra-cassandra-instance-node-1\" deleted\npersistentvolumeclaim \"var-lib-cassandra-cassandra-instance-node-2\" deleted\n\nstderr:\n"
time="2019-10-24T15:23:52+02:00" level=info msg="Successfully uninstalled 'cassandra' (instance='cassandra-instance', namespace='simple-install-uninstall-test-namespace')"
• [SLOW TEST:20.102 seconds]
simple-install-uninstall-test
/Users/murilo/mesosphere/kudo-cassandra-operator/tests/suites/simple_install_uninstall_test.go:35
  Uninstalls the operator
  /Users/murilo/mesosphere/kudo-cassandra-operator/tests/suites/simple_install_uninstall_test.go:44
------------------------------
time="2019-10-24T15:23:52+02:00" level=info msg="Deleting namespace 'simple-install-uninstall-test-namespace'"
time="2019-10-24T15:23:53+02:00" level=info msg="Deleted namespace 'simple-install-uninstall-test-namespace'"

Ran 2 of 2 Specs in 93.214 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 1m37.941930753s
Test Suite Passed
```

### CODE REVIEW

File: tests/utils/k8s/k8s.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+func DeleteNamespace(namespaceName string) error {
+	log.Infof("Deleting namespace '%s'", namespaceName)
+
+	err := clientset.CoreV1().Namespaces().Delete(
+		namespaceName,
+		&metav1.DeleteOptions{},
+	)
+	if err != nil {
+		log.Warnf("Error deleting namespace '%s': %s", namespaceName, err)
```
[the remark this answers is not in the document]
Comment (mpereira) [reply tests/utils/k8s/k8s.go]: Agree with returning and handling error vs using fatal, from a library perspective it makes more sense imo.

Regarding re-using namespaces, my idea is to create namespaces based on the test module name, like in the [simple-install-uninstall-test](https://github.com/mesosphere/kudo-cassandra-operator/blob/22a669bc0dd2e4a3017794f52b3031c44d961b28/tests/suites/simple_install_uninstall_test.go#L20-L22).

[the remark this answers is not in the document]
Comment (samvantran) [reply tests/utils/k8s/k8s.go]: @akirillov @rpalaznik, perhaps this approach should be done in kudo-spark?

[the remark this answers is not in the document]
Comment (mpereira) [reply tests/utils/k8s/k8s.go]: The general ideas for this testing structure are in the [proposal document](https://docs.google.com/document/d/1PhjMiwG318nPkehDZ4oEMSb6zcKcZeO-/edit) I shared earlier if you haven't seen it yet.

**Comment by mpereira** [comment]
Check failed on CI cluster teardown flake. Merging.

**Pull request merged**